### PR TITLE
Separate start and end depots

### LINKED ISSUE #604: Separate start and end depots
This has come up before (in #538, and originally in #305). Essentially, I propose we allow for the following:
```python
m = Model()
veh_type = m.add_vehicle_type(
    ...,
    start_depot=idx1,
    end_depot=idx2,
) 
```
This is a breaking change but it should make re-optimization a lot easier. Only question is how to deal with unused vehicles travelling between depots empty - do we count that towards the objective?

### PR DESCRIPTION
**Pull request by N-Wouda**
This PR:

- [x] Closes #604.
- [x] Adds and passes relevant tests.
- [x] Has well-formatted code and documentation.

<details>

**Notes**:

Please read our [contributing guidelines](https://pyvrp.org/dev/contributing.html) first.
In particular:

- You must add tests when making code changes.
  This keeps the code coverage level up, and helps ensure the changes work as intended.
- When fixing a bug, you must add a test that would produce the bug in the master branch, and then show that it is fixed with the new code. 
- New code additions must be well formatted. Changes should pass the pre-commit workflow, which you can set up locally using [pre-commit](https://pre-commit.com/#intro). 
- Docstring additions must render correctly, including escapes and LaTeX.
- Finally, it is essential that all contributions in this PR are license-compatible with PyVRP's MIT license.
  Please check that this PR can be included into PyVRP under the MIT license.

</details>

### CODE REVIEW

**Comment by N-Wouda** [comment]
@leonlan I think this is good to go now, could you have a look?

File: examples/quick_tutorial.ipynb [hunk 1]
```diff
@@ -330,7 +330,10 @@
     "\n",
     "Let's now solve a VRP with multiple depots and time windows.\n",
     "We consider two depots, and two vehicles per depot that have to start and end their routes at their respective depot.\n",
-    "For this, we will re-use some of the data from the VRPTW case, but change the time window data slightly: the first client now becomes the second depot."
+    "PyVRP additionally supports vehicles ending their routes at a different depot from where they start.\n",
```
[the remark this answers is not in the document]
Comment (leonlan) [reply examples/quick_tutorial.ipynb:333]:
```python
m = Model()

depots = [m.add_depot(x=COORDS[idx][0], y=COORDS[idx][1]) for idx in range(2)]

# Four vehicles, each with unique start and end depot combinations.
starts = [0, 1, 0, 1]
ends = [0, 1, 1, 0]

for start, end in zip(starts, ends):
    m.add_vehicle_type(
        1,
        start_depot=depots[start],
        end_depot=depots[end],
        max_duration=30,
        tw_early=TIME_WINDOWS[start][0],
        tw_late=TIME_WINDOWS[end][1],
    )
```

Just checked that this works.

**Review by leonlan — APPROVED** [review]

File: examples/quick_tutorial.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (N-Wouda) [reply examples/quick_tutorial.ipynb:333]: I'm going to make a new example that uses this, as well as some zone restrictions, for #615. That should hopefully be a bit more natural than simply zipping every possible combination.

File: pyvrp/plotting/plot_route_schedule.py
[the remark this answers is not in the document]
Comment (N-Wouda) [reply pyvrp/plotting/plot_route_schedule.py:1]: I've removed the depot time windows earlier. Could be that? I can update the reference images to be sure.

[the remark this answers is not in the document]
Comment (leonlan) [reply pyvrp/plotting/plot_route_schedule.py:1]: Yeah that’s probably it. I just checked on the SDVRP branch that this test also fail there.

**Pull request merged**